src/components/BlockchainList/PostsList.tsx

```diff
@@ -19,44 +21,56 @@ const scrollContainer = classnames(
 )
 
 function BlockchainPostsListSuspended() {
-  const { selectedPosts } = useSnapshot(PostStore)
+  const { selectedPosts, selectedToken, postsAmount, limit } =
+    useSnapshot(PostStore)
   const { selectedType } = useSnapshot(SelectedTypeStore)
+  const totalPosts = postsAmount[selectedType]
   const { hashStore, hashId } = useHashParams()
   const matchStore = hashStore && hashStore === selectedType
+  const [scrolledLimit, setScrolledLimit] = useState(limit)
 
-  const sliceToSpecificPost = (totalPosts: number) => {
-    if (!(matchStore || hashId)) return 10
-    return totalPosts - Number(hashId)
-  }
+  useEffect(() => {
+    function updateLimitIfNeeded() {
+      if (!(matchStore || hashId)) return
+      const neededLimit = totalPosts - Number(hashId)
+      setScrolledLimit(Math.max(neededLimit, PostStore.limit))
+      window.scrollTo({ top: window.innerHeight, behavior: 'smooth' })
+    }
 
-  const posts = PostStore.selectedToken
+    void updateLimitIfNeeded()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Hm, what if `params` will change or `limit` will change somehow? Maybe we shouldn't silence it? Maybe we can handle dependency array somehow? Also, how about to put it into helper?

Reply:
Seems like a part of `scrollToAnchor`